src/validate.ts

```diff
@@ -70,6 +70,17 @@ export function validate(
     return results;
   }
 
+  function collectEachBlockChild<T>(
+    node: ts.Block,
+    cb: (node: ts.Statement) => T[]
+  ): T[] {
+    const results: T[] = [];
+    ts.forEachChild(node, (child) => {
+      results.push(...cb(child as ts.Statement));
+    });
+    return results;
+  }
+
```

Comment:
Why make it specific to a block? This could be generic - collectEachChild

Reply:
The important part is that the callback acted on statements, not nodes. Updated to use `collectEachChild` internally.